eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/ResourceOperationMethodProvider.cs

```diff
@@ -123,6 +124,56 @@ protected virtual MethodSignature CreateSignature()
                 _convenienceMethod.Signature.NonDocumentComment);
         }
 
+        /// <summary>
+        /// Gets the method name for the operation, ensuring Create operations always use "CreateOrUpdate" name.
+        /// </summary>
+        /// <returns>The method name to use for this operation.</returns>
+        protected virtual string GetMethodNameForOperation()
+        {
+            if (IsCreateOperation())
+            {
+                return _isAsync ? "CreateOrUpdateAsync" : "CreateOrUpdate";
+            }
+            return _convenienceMethod.Signature.Name;
+        }
+
+        /// <summary>
+        /// Determines if the current operation is a Create operation.
+        /// </summary>
+        /// <returns>True if this is a Create operation, false otherwise.</returns>
+        protected virtual bool IsCreateOperation()
+        {
+            // Get the resource metadata to find the operation kind
+            var resourceMetadata = GetResourceMetadata();
+            if (resourceMetadata == null)
+                return false;
+
+            // Look up the operation by cross-language definition ID
+            foreach (var method in resourceMetadata.Methods)
+            {
+                if (method.Id == _serviceMethod.CrossLanguageDefinitionId && method.Kind == ResourceOperationKind.Create)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the resource metadata from the enclosing type.
+        /// </summary>
+        /// <returns>The resource metadata if available, null otherwise.</returns>
+        private ResourceMetadata? GetResourceMetadata()
+        {
+            // Access the resource metadata from the enclosing type
+            return _enclosingType switch
+            {
+                ResourceClientProvider resourceProvider => resourceProvider.ResourceMetadata,
+                ResourceCollectionClientProvider collectionProvider => collectionProvider.ResourceMetadata,
+                _ => null
+            };
+        }
```

Comment:
the logic here is quite complicated.
The reason why this is complicated is that we have to go back the `ResourceClientProvider` or `ResourceCollectionClientProvider` to get the resource metadata back.
How about we add a parameter on the ctor of this class which takes a `string? methodName = null`, and when constructing the signature in this class, we could use this as method name:
```
_methodName ?? _convenienceMethod.Signature.Name
```
And in the meantime, on the caller side of this type, we could write something to determine the method name from the operation kind (where we should have them, and we do not have to find a way to fetch them).